tritium-metrics/src/main/java/com/palantir/tritium/metrics/InternalCacheMetrics.java

```diff
@@ -40,12 +39,13 @@
 public final class InternalCacheMetrics {
     private InternalCacheMetrics() {}
 
-    public static <K> Map<K, Gauge<?>> createMetrics(Stats stats, Function<String, K> metricNamer) {
+    public static <K> ImmutableMap<K, Gauge<?>> createMetrics(Stats stats, Function<String, K> metricNamer) {
```

Comment:
@schlosna this was an ABI break, and is upsetting some internal products that use AtlasDB:

```
java.lang.NoSuchMethodError: com.palantir.tritium.metrics.InternalCacheMetrics.createMetrics(Lcom/palantir/tritium/metrics/InternalCacheMetrics$Stats;Ljava/util/function/Function;)Ljava/util/Map;
	at com.palantir.tritium.metrics.caffeine.CaffeineCacheStats.createCacheGauges(CaffeineCacheStats.java:75)
	at com.palantir.tritium.metrics.caffeine.CaffeineCacheMetrics.getMetrics(CaffeineCacheMetrics.java:46)
	at com.palantir.tritium.metrics.caffeine.CaffeineCacheStats.registerCache(CaffeineCacheStats.java:50)
	at com.palantir.atlasdb.util.AtlasDbMetrics.registerCache(AtlasDbMetrics.java:76)
	at com.palantir.atlasdb.cache.TimestampCache.<init>(TimestampCache.java:47)
	at com.palantir.atlasdb.factory.TransactionManagers.lambda$serializableInternal$12(TransactionManagers.java:404)
	at com.palantir.atlasdb.factory.TransactionManagers.initializeCloseable(TransactionManagers.java:561)
	at com.palantir.atlasdb.factory.TransactionManagers.serializableInternal(TransactionManagers.java:385)
	at com.palantir.atlasdb.factory.TransactionManagers.serializable(TransactionManagers.java:262)
	at com.palantir.atlasdb.factory.ImmutableTransactionManagers.access$2701(ImmutableTransactionManagers.java:43)
	at com.palantir.atlasdb.factory.ImmutableTransactionManagers$InitShim.serializable(ImmutableTransactionManagers.java:273)
	at com.palantir.atlasdb.factory.ImmutableTransactionManagers.<init>(ImmutableTransactionManagers.java:92)
	at com.palantir.atlasdb.factory.ImmutableTransactionManagers.<init>(ImmutableTransactionManagers.java:43)
	at com.palantir.atlasdb.factory.ImmutableTransactionManagers$Builder.build(ImmutableTransactionManagers.java:1075)
	at com.palantir.atlasdb.factory.TransactionManagers.createInMemory(TransactionManagers.java:252)
	at com.palantir.atlasdb.factory.TransactionManagers.createInMemory(TransactionManagers.java:236)
```

Reply:
Yeah, we need to publish a BOM (e.g. https://github.com/palantir/gradle-consistent-versions/pull/23 ) so that one does not pick up different versions of different tritium modules.

https://github.com/palantir/atlasdb/pull/4079 should also likely be merged to pick-up `com.palantir.tritium:* = 0.14.0`